otj-metrics-core/src/main/java/com/opentable/metrics/ready/Result.java

```diff
@@ -0,0 +1,168 @@
+     * @return a ready {@link Result} with an additional message
+     * @see String#format(String, Object...)
+     */
+    public static Result ready(String message, Object... args) {
+        return ready(String.format(message, args));
+    }
+
+    /**
+     * Returns an unready {@link Result} with the given message.
+     *
+     * @param message an informative message describing how the ready check failed
+     * @return an unready {@link Result} with the given message
+     */
+    public static Result unready(String message) {
+        return new Result(false, message, null);
+    }
+
+    /**
+     * Returns an unready {@link Result} with a formatted message.
+     * <p/>
+     * Message formatting follows the same rules as {@link String#format(String, Object...)}.
+     *
+     * @param message a message format
+     * @param args    the arguments apply to the message format
+     * @return an unready {@link Result} with an additional message
+     * @see String#format(String, Object...)
+     */
+    public static Result unready(String message, Object... args) {
+        return unready(String.format(message, args));
+    }
+
+    /**
+     * Returns an unready {@link Result} with the given error.
+     *
+     * @param error an exception thrown during the ready check
+     * @return an unready {@link Result} with the given {@code error}
+     */
+    public static Result unready(Throwable error) {
+        return new Result(false, error.getMessage(), error);
+    }
+
+    private final boolean ready;
+    private final String message;
+    private final Throwable error;
+    private final String timestamp =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX").format(ZonedDateTime.now());
+
+    public Result(boolean isReady, String message, Throwable error) {
+        this.ready = isReady;
+        this.message = message;
+        this.error = error;
+    }
+
+    /**
+     * Returns {@code true} if the result indicates the component is ready; {@code false}
+     * otherwise.
+     *
+     * @return {@code true} if the result indicates the component is ready
+     */
+    public boolean isReady() {
+        return ready;
+    }
+
+    /**
+     * Returns any additional message for the result, or {@code null} if the result has no
+     * message.
+     *
+     * @return any additional message for the result, or {@code null}
+     */
+    public String getMessage() {
+        return message;
+    }
+
+    /**
+     * Returns any exception for the result, or {@code null} if the result has no exception.
+     *
+     * @return any exception for the result, or {@code null}
+     */
+    public Throwable getError() {
+        return error;
+    }
+
+    /**
+     * Returns the timestamp when the result was created.
+     *
+     * @return a formatted timestamp
+     */
+    public String getTimestamp() {
+        return timestamp;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
```

Comment:
For equals, hashcode, and toString: I suggest using the Guava, Apache, or Immutables versions